### PR TITLE
Pin kube-state-metrics to last stable release branch

### PR DESCRIPTION
**Pull request by lilic**
Due to master going through breaking changes, using latest stable release branch release-1.9 is safest for users. Once release-2.0 branch is a bit more stable I plan on bumping it to that release.

cc @prometheus-operator/kube-prometheus-reviewers PTAL, thanks!

### CODE REVIEW

**Review by paulfantom — APPROVED** [review]
LGTM.

It is removing access to coordination.k8s.io/leases, but support for this is not implemented in KSM 1.9.7 so this is fine by me.

File: manifests/kube-state-metrics-deployment.yaml [hunk 1]
```diff
@@ -25,8 +25,6 @@ spec:
         - --telemetry-port=8082
         image: quay.io/coreos/kube-state-metrics:v1.9.7
         name: kube-state-metrics
-        securityContext:
```
Comment (zetaab) [inline manifests/kube-state-metrics-deployment.yaml:28]: why this was removed? Now our production does not work because of this

Comment (lilic) [reply manifests/kube-state-metrics-deployment.yaml:28]: @zetaab Hey you can always override and use master if that works for you or release-2.0. This is just in line with what is the release-1.9 kube-state-metrics branch, feel free to raise the PR fix there against the release-1.9 branch.

Comment (zetaab) [reply manifests/kube-state-metrics-deployment.yaml:28]: 👍  we did override for this to our jsonnet

**Pull request merged**